build/i2c/src/lib.rs

```diff
@@ -1770,3 +1747,33 @@ pub fn device_descriptions() -> impl Iterator<Item = I2cDeviceDescription> {
         },
     )
 }
+
+fn match_arms<'a, C>(
+    mut out: impl Write,
+    source: impl IntoIterator<Item = (&'a C, &'a Vec<usize>)>,
+    fmt: impl Fn(&C) -> String,
+) -> Result<()>
+where
+    C: 'a,
+{
+    for (controller, indices) in source {
+        let indices = indices
+            .iter()
+            .map(|&i| i..i + 1)
+            .collect::<RangeSet<usize>>();
+        let s: Vec<String> = indices
+            .iter()
+            .map(|range| format!("{}..={}", range.start, range.end - 1))
+            .collect::<_>();
```

Comment:
nit, take it or leave it: why not just put the `.join(...)` here and have `s` be a string rather than a vec of strings?

Reply:
Done!